src/client/renderer/r_occlude.c

```diff
@@ -101,43 +124,112 @@ _Bool R_CulludeSphere(const r_view_t *view, const vec3_t point, const float radi
 /**
  * @brief
  */
-void R_AddOcclusionQuery(r_view_t *view, const box3_t bounds) {
+r_occlusion_query_t R_CreateOcclusionQuery(const box3_t bounds) {
 
-	if (!r_occlude->integer) {
-		return;
-	}
+	r_occlusion_query_t query = {
+		.bounds = bounds,
+		.available = 1,
+		.result = 1,
+	};
 
-	if (R_CullBox(view, bounds)) {
-		return;
+	glGenQueries(1, &query.name);
+
+	// FIXME: This indexing is ghetto, but should be safe on most GL drivers
+	assert(query.name < MAX_OCCLUSION_QUERIES);
+	Box3_ToPoints(bounds, r_occlusion.vertexes[query.name]);
```

Comment:
@Paril Do you think this is an issue? Using the query object name as its index into an array? Every driver I've seen generates sequential object names from 1 onward. I'm doing this because I'd like to have linear storage for the query vertexes to simplify uploading and drawing them.